File: app/java/src/main/java/com/example/production/application/port/in/command/WorkOrderCreateCommand.java
package com.example.production.application.port.in.command;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;

/**
 * 作業指示作成コマンド
 */
@Data
@Builder
public class WorkOrderCreateCommand {
    private String orderNumber;
    private LocalDate workOrderDate;
    private String locationCode;
    private LocalDate plannedStartDate;
    private LocalDate plannedEndDate;
    private String remarks;
}

File: app/csharp/src/ProductionManagement.Application/Port/In/Command/WorkOrderCreateCommand.cs
namespace ProductionManagement.Application.Port.In.Command;

/// <summary>
/// 作業指示作成コマンド
/// </summary>
public class WorkOrderCreateCommand
{
    public required string OrderNumber { get; init; }
    public required DateOnly WorkOrderDate { get; init; }
    public required string LocationCode { get; init; }
    public required DateOnly PlannedStartDate { get; init; }
    public required DateOnly PlannedEndDate { get; init; }
    public string? Remarks { get; init; }
}
